flang/test/Lower/CUDA/cuda-stream.cuf

```diff
@@ -0,0 +1,15 @@
+! RUN: bbc -emit-hlfir -fcuda %s -o - | FileCheck %s
+
+attributes(global) subroutine sharedmem()
+  real, shared :: s(*)
+  integer :: t
+  t = threadIdx%x
+  s(t) = t
+end subroutine
+
+program test
+  call sharedmem<<<1, 1, 1024, 0>>>()
+end
+
+! CHECK-LABEL: func.func @_QQmain()
+! CHECK: cuf.kernel_launch @_QPsharedmem<<<%c1{{.*}}, %c1{{.*}}, %c1{{.*}}, %c1{{.*}}, %c1{{.*}}, %c1{{.*}}, %c1024{{.*}}, %{{.*}} : !fir.ref<i64>>>>()
```

Comment:
I wonder why the `stream` operand is made a reference while all the other operands are just values.  Maybe it will be more consitent to make it an optional `AnyInteger` operand, then you do not have to do anything special in lowering, and instead hide all the details of the kernel launch inside the `cuf.kernel_launch` convertion.

Just a thought.  I am not suggesting changing it in this PR.

Reply:
In CUDA Fortran stream is represented as a i64 integer. There is a special kind for it `cuda_stream_kind`. It can be written to by other API such as cudaStreamCreate not directly by `cuf.kernel_launch`.  So for consistency I think it is better to keep the restriction here. It is quite rare to pass the stream as a constant like in this test. Usually the user has a local variable and use the proper API to create the stream and pass it to the launch.

Reply:
I see.  To me the consistency of the dialect operation is more important than following to the letter the source APIs, so I would prefer just loading the value of stream before `cuf.kernel_launch`.  But this is just my preference :)

Thanks for the change!